app/uk/gov/hmrc/agentsubscriptionfrontend/service/AssuranceService.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscriptionfrontend.service
+
+import javax.inject.{Inject, Named, Singleton}
+
+import play.api.Logger
+import play.api.mvc.AnyContent
+import uk.gov.hmrc.agentmtdidentifiers.model.Utr
+import uk.gov.hmrc.agentsubscriptionfrontend.audit.AuditService
+import uk.gov.hmrc.agentsubscriptionfrontend.auth.AgentRequest
+import uk.gov.hmrc.agentsubscriptionfrontend.connectors.AgentAssuranceConnector
+import uk.gov.hmrc.agentsubscriptionfrontend.models._
+import uk.gov.hmrc.domain.{Nino, SaAgentReference, TaxIdentifier}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.frontend.auth.AuthContext
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class AssuranceService @Inject()(@Named("agentAssuranceFlag") agentAssuranceFlag: Boolean,
+                                 agentAssuranceConnector: AgentAssuranceConnector,
+                                 auditService: AuditService,
+                                 sessionStoreService: SessionStoreService) {
+
+  def assureIsAgent(utr: Utr)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[AssuranceResults]] = {
+    if (agentAssuranceFlag) {
+      agentAssuranceConnector.isManuallyAssuredAgent(utr).flatMap { isManuallyAssured =>
+        if(isManuallyAssured)
+          Future.successful(None)
+        else {
+          val futurePaye = agentAssuranceConnector.hasAcceptableNumberOfPayeClients
+          val futureSA = agentAssuranceConnector.hasAcceptableNumberOfSAClients
+
+          for {
+            hasAcceptableNumberOfPayeClients <- futurePaye
+            hasAcceptableNumberOfSAClients <- futureSA
+          } yield Some(AssuranceResults(hasAcceptableNumberOfPayeClients, hasAcceptableNumberOfSAClients))
```

Comment:
It might be better to have 'AssuranceResults' containing all decisions made, including manually assured and r2dw

Reply:
Sounds good, I'll change it.